File: web/src/api/modules/customerVisit.ts
import { request } from '../http'

export interface CustomerVisit {
  id: number
  operatorUserId: number
  operatorUserName?: string
  customerName: string
  inviter: string
  checkInLat: number
  checkInLng: number
  province: string
  city: string
  area: string
  detailAddress: string
  images: string
  visitPurpose: string
  remark: string
  visitDate: string
  createdAt: string
  updatedAt: string
}

export interface CustomerVisitListResponse {
  items: CustomerVisit[]
  total: number
  page: number
  pageSize: number
}

export interface CreateCustomerVisitRequest {
  customerName: string
  inviter?: string
  checkInLat: number
  checkInLng: number
  province?: string
  city?: string
  area?: string
  detailAddress: string
  images: string
  visitPurpose: string
  remark: string
}

export const getCustomerVisits = (params: {
  page?: number
  pageSize?: number
  keyword?: string
  startTime?: string
  endTime?: string
}) => {
  return request<CustomerVisitListResponse>({
    method: 'GET',
    url: '/v1/customer-visits',
    params,
  })
}

export const createCustomerVisit = (data: CreateCustomerVisitRequest) => {
  return request<{ id: number }>({
    method: 'POST',
    url: '/v1/customer-visits',
    data,
  })
}

export const uploadVisitImage = (file: File) => {
  const formData = new FormData()
  formData.append('file', file)
  return request<{ url: string }>({
    method: 'POST',
    url: '/v1/users/avatar/upload',
    data: formData,
  })
}

File: internal/model/customer_visit.go
package model

import "time"

// CustomerVisit 上门拜访签到记录
type CustomerVisit struct {
	ID               int64     `json:"id" gorm:"column:id;primaryKey;autoIncrement"`
	OperatorUserID   int64     `json:"operatorUserId" gorm:"column:operator_user_id;not null;index"`
	OperatorUserName string    `json:"operatorUserName,omitempty" gorm:"-"`
	CustomerName     string    `json:"customerName" gorm:"column:customer_name;not null"`
	Inviter          string    `json:"inviter" gorm:"column:inviter;not null;default:''"`
	CheckInIP        string    `json:"-" gorm:"column:check_in_ip;size:64;not null;default:''"`
	CheckInLat       float64   `json:"checkInLat" gorm:"column:check_in_lat;not null;default:0"`
	CheckInLng       float64   `json:"checkInLng" gorm:"column:check_in_lng;not null;default:0"`
	Province         string    `json:"province" gorm:"column:province;size:64;not null;default:''"`
	City             string    `json:"city" gorm:"column:city;size:64;not null;default:''"`
	Area             string    `json:"area" gorm:"column:area;size:64;not null;default:''"`
	DetailAddress    string    `json:"detailAddress" gorm:"column:detail_address;not null;default:''"`
	Images           string    `json:"images" gorm:"column:images;not null;default:'[]'"`
	VisitPurpose     string    `json:"visitPurpose" gorm:"column:visit_purpose;not null;default:''"`
	Remark           string    `json:"remark" gorm:"column:remark;not null;default:''"`
	VisitDate        string    `json:"visitDate" gorm:"column:visit_date;not null"`
	CreatedAt        time.Time `json:"createdAt" gorm:"column:created_at;autoCreateTime"`
	UpdatedAt        time.Time `json:"updatedAt" gorm:"column:updated_at;autoUpdateTime"`

	Operator *User `json:"-" gorm:"foreignKey:OperatorUserID;references:ID"`
}

func (CustomerVisit) TableName() string { return "customer_visits" }

// CustomerVisitCreateInput 创建上门拜访输入
type CustomerVisitCreateInput struct {
	OperatorUserID int64   `gorm:"-"`
	CustomerName   string  `gorm:"-"`
	Inviter        string  `gorm:"-"`
	CheckInIP      string  `gorm:"-"`
	CheckInLat     float64 `gorm:"-"`
	CheckInLng     float64 `gorm:"-"`
	Province       string  `gorm:"-"`
	City           string  `gorm:"-"`
	Area           string  `gorm:"-"`
	DetailAddress  string  `gorm:"-"`
	Images         string  `gorm:"-"`
	VisitPurpose   string  `gorm:"-"`
	Remark         string  `gorm:"-"`
	VisitDate      string  `gorm:"-"`
}

// CustomerVisitListFilter 上门拜访列表过滤
type CustomerVisitListFilter struct {
	OperatorUserID int64      `gorm:"-"`
	CanViewAll     bool       `gorm:"-"`
	Keyword        string     `gorm:"-"`
	StartTime      *time.Time `gorm:"-"`
	EndTime        *time.Time `gorm:"-"`
	Page           int        `gorm:"-"`
	PageSize       int        `gorm:"-"`
}

// CustomerVisitListResult 上门拜访列表结果
type CustomerVisitListResult struct {
	Items    []CustomerVisit `json:"items" gorm:"-"`
	Total    int64           `json:"total" gorm:"-"`
	Page     int             `json:"page" gorm:"-"`
	PageSize int             `json:"pageSize" gorm:"-"`
}
